src/healthcheck/HealthChecker.ts

```diff
@@ -203,48 +220,48 @@ class HealthChecker {
     let statusResponse: HealthStatus;
 
     // Handle liveness
-    statusResponse = new HealthStatus(State.UP)
+    statusResponse = new HealthStatus(State.UP);
 
     await Promise.all([this.getReadinessStatus(), this.getLivenessStatus()])
     .then((values) => {
-      let readiness = values[0]
-      let liveness = values[1]
+      let readiness = values[0];
+      let liveness = values[1];
 
       readiness.checks.map((check) => {
-        statusResponse.addStatus(check)
-      })
+        statusResponse.addStatus(check);
+      });
       liveness.checks.map((check) => {
-        statusResponse.addStatus(check)
-      })
-      return statusResponse
-    })
-    return statusResponse
+        statusResponse.addStatus(check);
+      });
+      return statusResponse;
+    });
+    return statusResponse;
   }
 
   private async getShutdownStatus(): Promise<HealthStatus>  {
     let statusResponse: HealthStatus;
 
     // Handle shutdown case
     if (this.shutdownRequested === true) {
-      statusResponse = new HealthStatus(State.STOPPING)
+      statusResponse = new HealthStatus(State.STOPPING);
       await this.shutdownPlugins.map((check) => {
         statusResponse.addStatus(check.getStatus());
-      })
-      return statusResponse
+      });
```

Comment:
It is suspicious. Please note the following code is executed synchronously and `check.runCheck()` is not invoked.

```ts
this.shutdownPlugins.map((check) => {
  statusResponse.addStatus(check.getStatus());
});
```

Reply:
Added
``` JS
const runChecks = await this.shutdownPlugins.map((check) => {
    check.runCheck();
 });
await Promise.all(runChecks);
await this.shutdownPlugins.forEach((check) => {
    statusResponse.addStatus(check.getStatus());
});
```